scripts/install-cni.sh

```diff
@@ -110,15 +115,23 @@ else
   cni_spec=${cni_spec//@cniBandwidthPlugin/}
 fi
 
-token=$(</var/run/secrets/kubernetes.io/serviceaccount/token)
 host=${KUBERNETES_SERVICE_HOST}
 # If host contains a colon (:), it is an IPv6 address, hence needs wrapping
 # with [..].
 if [[ "${host}" =~ : ]]; then
   host="[$host]"
 fi
-node_url="https://$host:${KUBERNETES_SERVICE_PORT}/api/v1/nodes/${HOSTNAME}"
-response=$(curl -k -s -H "Authorization: Bearer $token" "$node_url")
+node_url="https://$host:${KUBERNETES_SERVICE_PORT}/api/v1/watch/nodes/${HOSTNAME}"
+for attempt in {1..3}; do
+  log "Watching attempt #${attempt} at ${node_url}"
+  response=$(grep -m1 . <(curl -k -s -N -H "Authorization: Bearer $(</var/run/secrets/kubernetes.io/serviceaccount/token)" "$node_url" | jq --unbuffered -c '.object | select(.spec.podCIDR != null)'))
+  [[ -n "${response}" ]] && break
+done
+if [[ -z "${response}" ]]; then
+  fatal "Could not successfully watch node and wait for podCIDR."
+fi
+log "Node object fetched from $node_url"
+log "${response}"
```

Comment:
This whole block may be extracted into the bash function for better readability, i.e. get_k8s_node_descriptor. And additionally the variable name can be changed to something more meaningful than `$response`

Reply:
The response is somehow used throughout the file... don't bother to change it for now.

Reply:
Actually, it's not too many. Let me change it and get the opportunity to change and poor styles along the way.